core/src/main/java/org/jdbi/v3/BuiltInArgumentFactory.java

```diff
@@ -90,30 +123,55 @@ public Argument build(Class<?> expectedType, T value, StatementContext ctx)
 
         // Enums must be bound as VARCHAR.
         if (expectedType.isEnum()) {
-            return new StringArgument(value.toString());
+            return STR_BUILDER.build(value.toString());
         }
 
         // Fallback to generic ObjectArgument
-        return new ObjectArgument(value);
+        return OBJ_BUILDER.build(value);
```

Comment:
In fact, this duplication was even hiding a bug in plain sight!  Usage of `Enum#toString` vs `Enum#name`...